langchain/src/document_loaders/fs/chatgpt.ts

```diff
@@ -0,0 +1,103 @@
+import { TextLoader } from "./text.js";
+import { Document } from "../../document.js";
+
+interface ChatGPTMessage {
+  author: {
+    role: string;
+  };
+  content: {
+    parts: string[];
+  };
+  create_time: number;
+}
+
+interface ChatGPTLog {
+  title: string;
+  mapping: Record<string, { message: ChatGPTMessage }>;
+}
+
+function concatenateRows(message: ChatGPTMessage, title: string): string {
+  /**
+   * Combine message information in a readable format ready to be used.
+   * @param {ChatGPTMessage} message - Message to be concatenated
+   * @param {string} title - Title of the conversation
+   *
+   * @returns {string} Concatenated message
+   */
+  if (!message) {
+    return "";
+  }
+
+  const sender = message.author ? message.author.role : "unknown";
+  const text = message.content.parts[0];
+  const date = new Date(message.create_time * 1000)
```

Comment:
A bit more elegant to use e.g. `getMonth` and `getYear`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMonth

Reply:
Hi, I was wondering if you could elaborate a bit on what you mean by more elegant? Maybe I'm missing something, but since the loader timestamps messages with YYYY-MM-DD HH:MM:SS it seems like using `getYear`, `getMonth`, `getDate`, etc. would require more lines of code than the current implementation? Thanks

Reply:
Yeah no worries!